From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Mon, 3 Aug 2026 19:09:40 +0000
Subject: rockchip: rk3506: Update ENV_MEM_LAYOUT_SETTINGS

The RK3506G1 has 64 MiB integrated DDR2, the RK3506G2 has 128 MiB
integrated DDR3 and the RK3506B support external memory up to 1 GiB.

Current memory layout cause issues using compressed kernels or ramdisk
due to the default load addresses being used.

Change the default load addresses to also work on SoC variants with
limited memory, keep all default addresses in the 0-64 MiB range.

			 old		 new
  scriptaddr		5 MiB		2 MiB
  pxefile_addr_r	6 MiB		1 MiB
  kernel_addr_r		32.5 MiB	8 MiB
  kernel_comp_addr_r	128 MiB		48 MiB
  fdt_addr_r		30.5 MiB	4 MiB
  fdtoverlay_addr_r	31.5 MiB	6 MiB
  ramdisk_addr_r	96 MiB		48 MiB

Signed-off-by: Austin Lane <vidplace7@gmail.com>
---
 include/configs/rk3506_common.h | 14 +++++-----
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/include/configs/rk3506_common.h b/include/configs/rk3506_common.h
index 111111111111..222222222222 100644
--- a/include/configs/rk3506_common.h
+++ b/include/configs/rk3506_common.h
@@ -19,15 +19,15 @@
 #endif
 
 #define ENV_MEM_LAYOUT_SETTINGS			\
-	"scriptaddr=0x00500000\0"		\
+	"scriptaddr=0x00200000\0"		\
 	"script_offset_f=0xffe000\0"		\
 	"script_size_f=0x2000\0"		\
-	"pxefile_addr_r=0x00600000\0"		\
-	"kernel_addr_r=0x02080000\0"		\
-	"kernel_comp_addr_r=0x08000000\0"	\
-	"fdt_addr_r=0x01e00000\0"		\
-	"fdtoverlay_addr_r=0x01f00000\0"	\
-	"ramdisk_addr_r=0x06000000\0"		\
+	"pxefile_addr_r=0x00100000\0"		\
+	"kernel_addr_r=0x00800000\0"		\
+	"kernel_comp_addr_r=0x03000000\0"	\
+	"fdt_addr_r=0x00400000\0"		\
+	"fdtoverlay_addr_r=0x00600000\0"	\
+	"ramdisk_addr_r=0x03000000\0"		\
 	"kernel_comp_size=0x2000000\0"
 
 #define CFG_EXTRA_ENV_SETTINGS			\
-- 
Armbian

